terra/compute/base.py

```diff
@@ -186,5 +191,91 @@ def configuration_map_service(self, service_info):
 
     return service_info.volumes
 
+  @staticmethod
+  def configure_logger(sender, **kwargs):
+    if settings.terra.zone == 'controller':
+      print("SGR - setting up controller logging")
+
+      # Setup log file for use in configure
+      sender._log_file = os.path.join(settings.processing_dir,
+                                      terra.logger._logs.default_log_prefix)
+      os.makedirs(settings.processing_dir, exist_ok=True)
+      sender._log_file = open(sender._log_file, 'a')
+      sender.main_log_handler = StreamHandler(stream=sender._log_file)
+      sender.root_logger.addHandler(sender.main_log_handler)
+
+
+      # setup the TCP socket listener
+      sender.tcp_logging_server = LogRecordSocketReceiver(
+          settings.logging.server.hostname, settings.logging.server.port)
+      print('SGR - About to start TCP server...')
+
+      listener_thread = threading.Thread(
+          target=sender.tcp_logging_server.serve_until_stopped)
+      listener_thread.setDaemon(True)
+      listener_thread.start()
+
+      # Wait up to a second, to make sure the thread started
+      for _ in range(1000):
+        if sender.tcp_logging_server.ready:
+          break
+        time.sleep(0.001)
+
+      # Auto cleanup
+      @atexit.register
+      def cleanup_thread():
+        print("SGR - Sending cease and desist to LogRecordSocketReceiver")
+        sender.tcp_logging_server.abort = 1
+        listener_thread.join(timeout=5)
+        if listener_thread.is_alive():
+          print("SGR - LogRecordSocketReceiver thread did not die")
+        print("SGR - LogRecordSocketReceiver died!")
```

Comment:
These prints will be removed, but if something like them is kept, this print should be in the else